src/main/java/me/zeroeightsix/kami/module/modules/player/ViewLock.kt

```diff
@@ -0,0 +1,114 @@
+package me.zeroeightsix.kami.module.modules.player
+
+import me.zeroeightsix.kami.event.events.SafeTickEvent
+import me.zeroeightsix.kami.module.Module
+import me.zeroeightsix.kami.setting.Setting
+import me.zeroeightsix.kami.setting.Settings
+import me.zeroeightsix.kami.util.event.listener
+import java.util.*
+import kotlin.math.*
+
+@Module.Info(
+        name = "ViewLock",
+        category = Module.Category.PLAYER,
+        description = "Locks your camera view"
+)
+object ViewLock : Module() {
+
+    private val deltaXQueue: Queue<Pair<Int, Long>> = LinkedList<Pair<Int, Long>>(listOf())
+    private val deltaYQueue: Queue<Pair<Int, Long>> = LinkedList<Pair<Int, Long>>(listOf())
+    private var pitchSliceAngle = 1.0f
+    private var yawSliceAngle = 1.0f
+    val yaw = register(Settings.b("Yaw", true))
+    val pitch = register(Settings.b("Pitch", true))
+    val autoYaw = register(Settings.booleanBuilder("AutoYaw").withValue(true).withVisibility { yaw.value })
+    val autoPitch = register(Settings.booleanBuilder("AutoPitch").withValue(true).withVisibility { pitch.value })
+    private val threshold = register(Settings.integerBuilder("ChangeThresh").withValue(5).withRange(1, 50).withStep(1).withVisibility { autoPitch.value || autoYaw.value })
+    val disableMouseYaw = register(Settings.booleanBuilder("DisableMouseYaw").withValue(true).withVisibility { yaw.value })
+    val disableMousePitch = register(Settings.booleanBuilder("DisableMousePitch").withValue(true).withVisibility { pitch.value })
+    private val specifcYaw = register(Settings.floatBuilder("SpecificYaw").withValue(180.0f).withRange(-180.0f, 180.0f).withStep(1.0f).withVisibility { !yaw.value })
+    private val specificPitch = register(Settings.floatBuilder("SpecificPitch").withValue(0.0f).withRange(-90.0f, 90.0f).withStep(1.0f).withVisibility { !pitch.value })
+    private val yawSlice = register(Settings.integerBuilder("YawSlice").withValue(8).withRange(2, 32).withStep(1).withVisibility { autoYaw.value })
+    private val pitchSlice = register(Settings.integerBuilder("PitchSlice").withValue(5).withRange(2, 32).withStep(1).withVisibility { autoPitch.value })
+
+    private var yawSanp = 0
+    private var pitchSnap = 0
+
+
+    override fun onEnable() {
+        yawSliceAngle = 360.0f / yawSlice.value
+        pitchSliceAngle = 180.0f / (pitchSlice.value - 1)
+        if ((autoYaw.value || autoPitch.value) && mc.player != null)
+            snapToNext()
+    }
+
+    init {
+        listener<SafeTickEvent> {
+            if (autoYaw.value || autoPitch.value) {
+                snapToSlice()
+
+            } else {
+                if (yaw.value) mc.player.rotationYaw = specifcYaw.value
+                if (pitch.value) mc.player.rotationPitch = specificPitch.value
+            }
+        }
+        yawSlice.settingListener = Setting.SettingListeners {
+            if (isEnabled && autoYaw.value) snapToNext()
+            yawSliceAngle = 360.0f / yawSlice.value
```

Comment:
Have a empty line between the listener and the setting listener to make it more clear. Or you can have all the setting listeners in another init block at the bottom of the class